### PR TITLE
sinking block benchmark

### PR DESCRIPTION
**Pull request by cedrict**
This is a benchmark needed for the Q1-Q1 paper Wolfgang and I are writing. I meant to do this during the hackathon but never came to do it.

### CODE REVIEW

File: benchmarks/sinking_block/sinking_block.prm [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    set Y extent  = 512e3
+  end
+end
+
+subsection Boundary velocity model
+  set Tangential velocity boundary indicators = left, right, bottom, top
+end
+
+#subsection Discretization 
+#   set Stokes velocity polynomial degree       = 1
```
[the remark this answers is not in the document]
Comment (gassmoeller) [reply benchmarks/sinking_block/sinking_block.prm]: If not then leave a comment above, like `for testing the Q1Q1 element uncomment the following lines`

**Comment by cedrict** [comment]
hopefully addressed all comments and did my test right :)

File: benchmarks/sinking_block/sinking_block.cc [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                                 Patterns::Double (0),
+                                 "Viscosity in the Inclusion.");
+              prm.declare_entry ("rho1", "3200",
+                                 Patterns::Double (0),
+                                 "density of the mantle.");
+              prm.declare_entry ("rho2", "3232",
+                                 Patterns::Double (0),
+                                 "density in the Inclusion.");
+              prm.declare_entry ("method", "0",
+                                 Patterns::Integer (0),
```
Comment (tjhei) [inline benchmarks/sinking_block/sinking_block.cc]:
```suggestion
                                 Patterns::Integer (0,2),
```

File: benchmarks/sinking_block/sinking_block.cc [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        {
+          prm.enter_subsection("Material model");
+          {
+            prm.enter_subsection("SinkingBlock");
+            {
+              eta1 = prm.get_double ("eta1");
+              eta2 = prm.get_double ("eta2");
+              rho1 = prm.get_double ("rho1");
+              rho2 = prm.get_double ("rho2");
+              method = prm.get_double ("method");
```
Comment (tjhei) [inline benchmarks/sinking_block/sinking_block.cc]:
```suggestion
              method = prm.get_integer ("method");
```

File: benchmarks/sinking_block/sinking_block.prm (hunk 1 above)
[the remark this answers is not in the document]
Comment (cedrict) [reply benchmarks/sinking_block/sinking_block.prm]: since we don't wish people to use Q1Q1, I have removed the line from the input file.

**Comment by cedrict** [comment]
I believe I have addressed all above comments. Also added a README.md

**Comment by tjhei** [comment]
documentation doesn't build:
``
! Undefined control sequence.
l.9165 $|{\bm
              g}|=10\text{m}/\text{s}^2$. Boundary conditions are free slip ...
``
You might need to rebase...

**Comment by cedrict** [comment]
the compilation error was due to my using \bm instead of \boldsymbol.

**Comment by tjhei** [comment]
you will need to update the test results: https://jenkins.tjhei.info/job/aspect/job/PR-3146/5/artifact/changes-test-results.diff

**Comment by gassmoeller** [comment]
I took the liberty of updating the test results and simplified the test .prm by using the benchmark file. Since everything else was reviewed already this should be ready if the tester is happy.

**Comment by cedrict** [comment]
oh thanks a bunch! :)

**Comment by tjhei** [comment]
great!

**Comment by gassmoeller** [comment]
No problem, just trying to reduce the number of open issues and pull requests :-)

**Pull request merged**